File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    dict ={}
    sumstr=''
    arrnum = ['0','1','2','3','4','5','6','7','8','9']
    answer = ''
    dict ={
    '0':'zero',
    '1':'one',
    '2':"two",
    '3':"three",
    '4':'four',
    '5':'five',
    '6':'six',
    '7':'seven',
    '8':'eight',
    '9':'nine'
    }

    for i in range(len(s)) :
        if s[i] in arrnum :
            answer += s[i]
        elif sumstr in dict.values() :
            for k, v in dict.items():
                if v == sumstr :
                    answer+=k
                    sumstr=''
        else :
            sumstr+=s[i]
            for k, v in dict.items():
                if v == sumstr :
                    answer+=k
                    sumstr=''

    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.HashMap;

class Solution {
    public int solution(String s) {
        String answer = "";
        HashMap <String, Integer> map = new HashMap<>();
        map.put("zero",0);
        map.put("one",1);
        map.put("two",2);
        map.put("three",3);
        map.put("four",4);
        map.put("five",5);
        map.put("six",6);
        map.put("seven",7);
        map.put("eight",8);
        map.put("nine",9);
        
        String pt = "";
        
        for(int i=0; i<s.length(); i++){
            if(Character.isDigit(s.charAt(i))){
                answer += s.charAt(i);
            }
            else{
                pt+= s.charAt(i);
                if (map.containsKey(pt)){
                    String a = String.valueOf(map.get(pt));
                    answer += a;
                    pt ="";
                }
            }
        }
        
        return Integer.parseInt(answer);
    }
}